### DISCUSSION TITLE
Translation strategies

### DISCUSSION THREAD

**Discussion by awvwgk** [fortran-lang/fpm-docs#33] [General]
What is your strategy for translating? Any tools you did find useful? Also, how do you like the po-files for handling the translations so far?

I'm currently directly translating by opening the generated po-files with [neovim](https://neovim.io), which I found quite convenient once I figured out that [GitHub's copilot](https://copilot.github.com/) can also suggest decent automatic translations (grammar is okayish, cases and gender are often wrong). On average those seem worse than what you would get with a tool like [deepl](https://www.deepl.com/translator) (based on a similar ML strategy), but they are directly available in my editor.

*Copilot suggestion in line 40 in light gray:*
![image](https://user-images.githubusercontent.com/28669218/146642544-f4458c2b-29a0-4f30-8ab9-49e8c298f117.png)

I also tried a translation editor ([gtranslator](https://wiki.gnome.org/Apps/Gtranslator)), but didn't really found this advantageous yet.

**Comment by vmagnin** [comment]
It can be useful to have a look to the TOML site to fully understand the vocabulary used for describing the fpm manifest: https://toml.io/en/v1.0.0
Note that this site is translated in several Asiatic languages and also in French: https://toml.io/fr/v1.0.0
which can help for translating the fpm-docs.

**Comment by vmagnin** [comment]
I am using the [poedit](https://poedit.net/) tool available for all platforms. For small sentences I am directly translating in French. For long sentences or paragraphs, I am using the [DeepL](https://www.deepl.com/) free online translating tool to accelerate the work. I am very impressed by the result (do you remember those tools in the 2000's?). I verify the translation and change a few words or expressions for improvement when needed, then I copy back the result in poedit.

Some useful keyboard shortcuts in poedit:

- CTRL+B to copy the English text as is,
- CTRL+down arrow to go to the next item.

poedit is managing the length of the lines (79 characters by default). And it displays some stats so you know how much work is still to be done.

Finally, a problem with translation is that you tend to focus on each sentence or paragraph, so it's difficult to keep in mind the whole picture. When the translation is finished, you need to print everything and calmly read it to detect the problems and improve your work.

**Comment by certik** [comment]
I use neovim as well.

I am also testing Weblate, and I'll update here once I learn how it works.
